e2e/cloud.go

```diff
@@ -0,0 +1,113 @@
+package e2e_test
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+	"github.com/Azure/go-armbalancer"
+)
+
+type azureClient struct {
+	coreClient     *azcore.Client
+	vmssClient     *armcompute.VirtualMachineScaleSetsClient
+	vmssVMClient   *armcompute.VirtualMachineScaleSetVMsClient
+	resourceClient *armresources.Client
+	aksClient      *armcontainerservice.ManagedClustersClient
+}
+
+func newAzureClient(subscription string) (*azureClient, error) {
+	httpClient := &http.Client{
+		// For Now using the defaults recommended by Track 2
```

Comment:
Should we add a link to the doc that mentions the recommended defaults? In case the defaults change

Reply:
I don't actually know where these are documented but they're relatively sane compared to many stdlib defaults

avoid slow loris, avoid indefinite timeouts, etc

Reply:
will comment